flask_webapp/tests/test_bleConnectAndSend.py

```diff
@@ -0,0 +1,42 @@
+import time
+from bluepy import *
+
+# Bluetooth Low Energy communication test
+# The function below sends a command to a specific ble device.
+# To connect the ble device, the function will require the ble
+# device's bluetooth address & the message as a byte value. The
+# message will be sent byte by byte.
+def sendMsg(bdaddr,msg_byte):
+    print("Connecting...")
+    
+    # Using the bluepy library we create a "Peripheral" object with the 
+    # bluetooth address. 
+    # NOTE : BLE connections work as a Client (master) and 1 or more 
+    # Servers (slaves)
+    dev = btle.Peripheral(bdaddr)
+
+    # BLE devices use services to handle different tasks to communicate
+    # different data types (i.e. heart rate monitoring or pace makers).
+    # Each service has multiple charecteristics. Below we get the correct
+    # service for the HM-10 ble module on the arduino, and write it with
+    # the characteristic object.
+    services = list(dev.services)
+    characteristic = services[len(services)-1].getCharacteristics()[0]
```

Comment:
You can do ```services[-1]``` for the last element in a list.

Reply:
can services be an empty list?

Reply:
services will never be an empty list based on the Bluetooth Specification. Every BLE devices must have services.
[BT SPEC](https://www.bluetooth.com/specifications/bluetooth-core-specification/)